lib/matplotlib/tests/test_ticker.py

```diff
@@ -511,37 +511,223 @@ def test_formatstrformatter():
     assert '002-01' == tmp_form(2, 1)
 
 
-percentformatter_test_cases = (
-    # Check explicitly set decimals over different intervals and values
-    (100, 0, '%', 120, 100, '120%'),
-    (100, 0, '%', 100, 90, '100%'),
-    (100, 0, '%', 90, 50, '90%'),
-    (100, 0, '%', 1.7, 40, '2%'),
-    (100, 1, '%', 90.0, 100, '90.0%'),
-    (100, 1, '%', 80.1, 90, '80.1%'),
-    (100, 1, '%', 70.23, 50, '70.2%'),
-    # 60.554 instead of 60.55: see https://bugs.python.org/issue5118
-    (100, 1, '%', 60.554, 40, '60.6%'),
-    # Check auto decimals over different intervals and values
-    (100, None, '%', 95, 1, '95.00%'),
-    (1.0, None, '%', 3, 6, '300%'),
-    (17.0, None, '%', 1, 8.5, '6%'),
-    (17.0, None, '%', 1, 8.4, '5.9%'),
-    (5, None, '%', -100, 0.000001, '-2000.00000%'),
-    # Check percent symbol
-    (1.0, 2, None, 1.2, 100, '120.00'),
-    (75, 3, '', 50, 100, '66.667'),
-    (42, None, '^^Foobar$$', 21, 12, '50.0^^Foobar$$'),
-)
-
-
 @pytest.mark.parametrize('xmax, decimals, symbol, x, display_range, expected',
```

Comment:
this is purely style, but I prefer to leave the test_cases and id out of the decorator (the way you had before) and just pass in the variables (just as you had) instead of creating the super long decorator)
here's a teeny example: https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/tests/test_category.py#L17

but I am thrilled about the ids